docs/source/data/resolvers.mdx

```diff
@@ -461,6 +461,14 @@ type Author {
 
 If the resolver for the `books` field returns an array of objects that each contain a `title` field, then you can use a default resolver for the `title` field. The default resolver will correctly return `parent.title`.
 
+## Resolving types
+
+There are GraphQL types that enable you to define a field that returns one of multiple possible object types (i.e. [unions and interfaces](../schema/unions-interfaces/#resolving-a-union)). To resolve a field that can return different object types, you must define a `__resolveType` function to inform Apollo Server _which_ object type to return.
```

Comment:
Worth noting that (while our docs don't seem to document this, and while it's probably a good idea to do `__resolveType` anyway) there actually is a default `__resolveType` if you don't pass on for a given abstract type. It checks to see if the in-memory representation of the object is an "object" with a `__typename` field and if so, it uses the value of that field.  If not, it lets you also use a different approach that we also don't document — instead of putting `__resolveType(thingOfUnknownType) => string` on a union (or interface) type, you can put `__isTypeOf(thingOfUnknownType) => boolean` on each of the object types in the union/interface, and the default `__resolveType` will call them all and choose one that returns true.

Reply:
Oooh, that's really interesting about `__isTypeOf`! After doing a little digging, it seems like `__isTypeOf` would be helpful to document, showing how to break up Unions/Interfaces that include a lot of different object types (so you can break up the large "if..else" block that  `__resolveType` would require). 

@glasser  If you agree it's valuable, I can go ahead and add an additional section on `__isTypeOf`  to the "Unions and Interfaces" article.

Reply:
I will handle documenting this in a separate PR!  :)